src/test/java/com/avispl/symphony/dal/communicator/other/genericserver/WebClientCommunicatorTest.java

```diff
@@ -0,0 +1,204 @@
+	public void init() throws Exception {
+		wireMockRule.start();
+		webClientCommunicator = new WebClientCommunicator();
+		webClientCommunicator.setTrustAllCertificates(true);
+		webClientCommunicator.setProtocol(PROTOCOL);
+		webClientCommunicator.setPort(PORT);
+		webClientCommunicator.setHost(HOST_NAME);
+		webClientCommunicator.setContentType("text/plain");
+		webClientCommunicator.setAuthenticationScheme(HttpCommunicator.AuthenticationScheme.None);
+		webClientCommunicator.init();
+	}
+
+	@AfterEach
+	public void stopWireMockRule() {
+		webClientCommunicator.destroy();
+		wireMockRule.stop();
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with response status code 200 OK for HTML content
+	 */
+	@Test
+	public void getMultipleStatisticsWithHtmlPage() throws Exception {
+		// Attempt to check accessible for index.html
+		webClientCommunicator.setURI("/");
+		ExtendedStatistics extendedStatistics = (ExtendedStatistics) webClientCommunicator.getMultipleStatistics().get(0);
+		Map<String, String> stats = extendedStatistics.getStatistics();
+		assertEquals("200 OK", stats.get("URI Status"));
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with response status code 200 OK for XML file
+	 */
+	@Test
+	public void getMultipleStatisticsWithXMLFile() throws Exception {
+		// Attempt to check accessible for a xml file
+		webClientCommunicator.setURI("/xml-file.xml");
+		ExtendedStatistics extendedStatistics = (ExtendedStatistics) webClientCommunicator.getMultipleStatistics().get(0);
+		Map<String, String> stats = extendedStatistics.getStatistics();
+		assertEquals("200 OK", stats.get("URI Status"));
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with response status code 200 OK for Jpg file
+	 */
+	@Test
+	public void getMultipleStatisticsWithJpgFile() throws Exception {
+		// Attempt to check accessible for a jpg file
+		webClientCommunicator.setURI("/jpg-file.jpg");
+		ExtendedStatistics extendedStatistics = (ExtendedStatistics) webClientCommunicator.getMultipleStatistics().get(0);
+		Map<String, String> stats = extendedStatistics.getStatistics();
+		assertEquals("200 OK", stats.get("URI Status"));
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with response status code 403 Forbidden.
+	 */
+	@Test
+	public void getMultipleStatisticsWithStatusCode403() throws Exception {
+		//  Expect throw API Error
+		webClientCommunicator.setURI("/forbidden");
+		ExtendedStatistics extendedStatistics = (ExtendedStatistics) webClientCommunicator.getMultipleStatistics().get(0);
+		Map<String, String> stats = extendedStatistics.getStatistics();
+		assertEquals("403 Forbidden", stats.get("URI Status"));
+	}
+
+	/**
+	 * Test method for {@link WebClientCommunicator#getMultipleStatistics()} with API Error due to 404 response status code.
+	 */
+	@Test
+	public void getMultipleStatisticsWithStatusCode404() {

```

Comment:
if you expect getMultipleStatistics() to throw an error here, consider using 
`@Test(expected = SomeException.class)`